mlrun/utils/helpers.py

```diff
@@ -1371,6 +1371,20 @@ def has_timezone(timestamp):
         return False
 
 
+def format_datetime(dt: datetime) -> str:
+    if dt.tzinfo is None:
+        dt = dt.replace(tzinfo=timezone.utc)
+
+    # Convert to UTC timezone
+    with_timezone = dt.astimezone(timezone.utc)
+
+    # Format as requested
+    formatted_time = with_timezone.strftime("%Y-%m-%d %H:%M:%S.%f%z")
+
+    # Replace the "+" sign with the ":"
+    return formatted_time[:-2] + ":" + formatted_time[-2:]
```

Comment:
on python 3.12 you could use `%:z`. so this thingy is temporary until python 3:12 become minimal version.
in addition to that, you dont replace the "+" - you add ":" between the hours and minutes of timezone

also, your function overrides datetime timezone. for when datetime is naive - I can understand. but why for other use cases?